packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -704,11 +711,16 @@ export class GraphQLApi extends Construct {
       definition = '';
     } else if ( this.schemaMode == SchemaDefinition.CODE && file) {
       throw new Error('definition mode CODE is incompatible with file definition. Change mode to FILE/S3 or unconfigure schemaDefinitionFile');
+    } else if ( this.schemaMode == SchemaDefinition.S3 && !file) {
+      throw new Error('schemaDefinitionFile must be configured if using S3 definition mode.');
+    } else if ( this.schemaMode == SchemaDefinition.S3 && file ) {
+      definitionS3Location = file;
```

Comment:
```suggestion
    } else if ( this.schemaMode === SchemaDefinition.S3 && !file) {
      throw new Error('schemaDefinitionFile must be configured if using S3 definition mode.');
    } else if ( this.schemaMode === SchemaDefinition.S3 && file ) {
      definitionS3Location = file;
```

seems like this needs to be fixed up in this entire conditional block